[Gabriel Yang] iP

gabyang opened the pull request:
## DukePro

>"Forgive my manners, call me the Duke. Now to business. Weapons, ammunition, healing salves – anything you desire, I can provide." - [The Duke Resident Evil](https://residentevil.fandom.com/wiki/The_Duke/quotes)

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is:

1. download it [here](https://github.com/nus-cs2103-AY2223S1)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

``
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
``

Commit:
Add Gradle support

Commit:
Completed Level-1

Commit:
Complete Level-2

Commit:
Complete Level-3

Commit:
Complete Level-4

Commit:
text-ui-test: added tests

Commit:
Complete Level-5

Commit:
Complete Level-6

Commit:
Added Arraylist improvement

Commit:
Added DukeTaskList file

Commit:
Abstracted time field in DukeTask

Commit:
Add Read and Write function to file

Commit:
Remove Commented code

Commit:
Merge branch 'master' into Level-7

Commit:
Revert to before Level-7 Merge due to error

Commit:
Complete Level-7 (Redo)

Commit:
Merge branch 'branch-Level-7-redo'

Commit:
Add Dates and Times formatting

Commit:
Merge branch 'branch-Level-8'

Commit:
Merge branch 'add-gradle-support'

Commit:
Add More Classes for DukeTask

Commit:
Add more classes for DukeTask files

Commit:
Add Classes to improve OOP design

Commit:
Add Packages for all classes

Commit:
Update Gradle

Commit:
Add JUnit Tests for DukeTaskEvent and DukeTask

Commit:
Add JavaDoc

Commit:
Remove unecessary comments

Commit:
Complete Level-9

Commit:
Merge branch 'A-JavaDoc'

Commit:
Merge branch 'A-CodingStandard'

Commit:
Merge tag 'Level-9'

HoJunHao2000 commented on src/main/java/command/ByeCommand.java:
```diff
@@ -0,0 +1,26 @@
+ */
+public class ByeCommand extends Command {
+    /**
+     * Does Not do Anything
```
- Do leave a line separating the description and parameters. 
- The code does something not nothing. 
- Do add a description for each parameter too! 
- All sentences should end with .
``` java
/**
 * Prints the exit message.
 *      
 * @param taskList
 * @param ui
 * @param storage
 * @throws InvalidFormatException
 */
```

gabyang replied:
Got it thanks!

HoJunHao2000 commented on src/main/java/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,54 @@
+     * @throws InvalidFormatException
+     */
+    @Override
+    public void deconstruct(ArrayList<DukeTask> tasklist, Ui ui, Storage storage) throws InvalidFormatException {
```
- Do use OOP principles and separate components of Ui, Parser, and Commands apart from each other
- Consider moving all print statements into the Ui class. You have access to it through the ui argument in deconstruct
- Parsing should be done before passing into the Command object subclasses

gabyang replied:
Ok definitely!

HoJunHao2000 commented on src/main/java/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,54 @@
+ * Represents a deadline command
+ */
+public class DeadlineCommand extends Command {
+    private String cmd;
```
Try to use meaning full variable names instead. E.g. command. Should only use shortened names if names are too long or for temporary variables (e.g. in for loops)

gabyang replied:
I'll make note of that!

HoJunHao2000 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,56 @@
+     */
+    public void run() {
+        storage.read();
+        Scanner input = new Scanner(System.in);
```
you can consider moving your scanner into ui

gabyang replied:
Yep

HoJunHao2000 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,53 @@
+package duke;
+
+import command.*;
```
Do not use wildcard imports. List all imports down individually.

gabyang replied:
Ok!

HoJunHao2000 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,93 @@
+    private static String FILEPATH;
+    private static ArrayList<DukeTask> tasklist;
+    
+    Storage() {}
```
Unneccessary

HoJunHao2000 commented on src/main/java/task/DukeTask.java:
```diff
@@ -0,0 +1,61 @@
+    private boolean isMarked;
+    private char taskType;
+
+    public DukeTask () {
```
Missing javadocs for many public methods

HoJunHao2000 commented on src/main/java/task/DukeTaskDeadline.java:
```diff
@@ -0,0 +1,42 @@
+ */
+public class DukeTaskDeadline extends DukeTask {
+    
+    private LocalDateTime ldt;
```
use meaningful names

HoJunHao2000 reviewed (commented):
- Alot of works need to be done on the javadocs
- try to use meaningful names for the variables
- use OOP concepts

ThomasHoooo commented:
Hey awesome application! Can't wait to try out your product!

Commit:
Add Basic UI code

Commit:
Add GUI for Duke

Commit:
Merge branch 'branch-Level-10'

Commit:
Refactor code for Duke, DialogBox and UI

Commit:
Refactor Commands to return String and fix bugs

gabyang committed:
Update README.md

Commit:
Add Assertions

gabyang committed:
Merge pull request #1 from gabyang/A-Assertions

Add Assertion statements

Commit:
Make Code less nested

gabyang committed:
Merge pull request #2 from gabyang/A-CodeQuality

Make Code less nested

Commit:
Add functionality to Find: find by task number
Find Command needs to do more than just find tasks.
Users want to also find tasks by the number in order to prioritise
which task they want to do first.

Commit:
Add functionality to find: find by task type
Find command needs to do more than just find task by tasks or
by task number.
Users also want to find tasks by task type in order to filter out
tasks that have a dateline and those that is not constrained by time.

Commit:
Merge tag 'BCD-Extension'

Commit:
Merge branch 'master' of https://github.com/gabyang/ip

Commit:
Integrate JavaFXML

Commit:
New Changes for Debugging purposes

Commit:
Merge branch 'Integrate-JavaFXML'

Commit:
Clean up Program

Commit:
Update UserGuide

Commit:
Add pictures of UI

Commit:
Update UserGuide

Commit:
Delete Unused files and code

gabyang committed:
Update UserGuide, Fix Formatting

Commit:
Format JavaDocs and Fix bugs

Commit:
Make DeadlineCommand more user friendly

Commit:
Update UserGuide Fix bug for missing list.txt

gabyang committed:
Update README.md
